lib/apiservers/engine/backends/container.go

```diff
@@ -1336,7 +1336,36 @@ func (c *Container) ContainerLogs(ctx context.Context, name string, config *back
 // ContainerStats writes information about the container to the stream
 // given in the config object.
 func (c *Container) ContainerStats(ctx context.Context, name string, config *backend.ContainerStatsConfig) error {
-	return fmt.Errorf("%s does not yet implement ContainerStats", ProductName())
+	defer trace.End(trace.Begin(name))
+
+	// Look up the container name in the metadata cache to get long ID
+	vc := cache.ContainerCache().GetContainer(name)
+	if vc == nil {
+		return NotFoundError(name)
+	}
+
+	// get the configured CPUMhz for this VCH so that we can calculate docker CPU stats
+	cpuMhz, err := systemBackend.SystemCPUMhzLimit()
+	if err != nil {
+		log.Errorf("unable to gather systemCPUMhz for container(%s): %s", vc.ContainerID, err)
+		return InternalServerError("unable to gather container statistics")
+	}
+
+	out := config.OutStream
+	if config.Stream {
+		// Outstream modification (from Docker's code) so the stream is streamed with the
+		// necessary headers that the CLI expects.  This is Docker's scheme.
+		wf := ioutils.NewWriteFlusher(config.OutStream)
+		defer wf.Close()
+		wf.Flush()
```

Comment:
Not sure..was just following the same thing done in ContainerLogs -- seems to work fine.